server/storage/mvcc/kvstore_compaction.go

```diff
@@ -48,24 +52,27 @@ func (s *store) scheduleCompaction(compactMainRev int64) error {
 
 		tx := s.b.BatchTx()
 		tx.LockOutsideApply()
-		keys, _ := tx.UnsafeRange(schema.Key, last, end, int64(batchNum))
-		for _, key := range keys {
-			rev = bytesToRev(key)
+		keys, values := tx.UnsafeRange(schema.Key, last, end, int64(batchNum))
```

Comment:
Please see that `end` is defined as: 
```
binary.BigEndian.PutUint64(end, uint64(compactMainRev+1))
```

I'm afraid this logic would miss all entries created after the most recent compaction... but maybe I'm missing sth...

Reply:
`compactMainRev` is revision that is being compacted. This is our goal, calculate hash at the same time as compaction is being done, thus not increasing number of times entry is touched. We don't care about entries after most recent compaction.